core/lib/spree/event/processors/mailer_processor.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+module Spree
+  module Event
+    module Processors
+      module MailerProcessor
```

Comment:
Would it be worth providing a registration interface? Something like:
```ruby
module Spree
  module EventRegistration
    def subscriptions
      @subscriptions ||= []
    end

    def add_subscription(method_name = nil)
      subscriptions << method_name.to_sym
      send(:mattr_accessor, "#{method_name}_handle")
      subscriptions
    end

    def register!
      subscriptions.each do |subscription|
        instance_variable_set("@#{subscription}_handle", send(:subscription))
      end
    end

    def unregister!
      subscriptions.each do |subscription|
        Spree::Event.unsubscribe(send("#{subscription}_handle"))
      end
    end
  end
end
```

Then in any processor

```ruby
module Spree
  module Event
    module Processors
      module MailerProcessor
        extend self
        extend Spree::EventRegistration
 
        add_subscription :order_finalize
        add_subscription :reimbursement_perform

        def order_finalize
          # Do work
        end

        def reimbursement_perform
          # Do work
        end
      end
    end
  end
end
```

It's a bit much to look at, but may allow for more extensibility as well as shareability.

Reply:
I know this probably isn't the route, but when obtaining a handle to subscribers it may be worth implementing a consistent methodology so that store may unsubscribe from out of the box subs.

Reply:
I think this is a good concern, we should provide one suggested way to do things that we know will be needed in stores and extensions.

Reply:
@skukx 👍 thanks for the valuable input, I'll be working on this!